File: frontend/src/types/temporal.ts
export interface HorizonForecastItem {
  horizon_years: number
  forecast_year: number
  predicted_yield: number
  lower_bound_p10: number
  upper_bound_p90: number
  prediction_spread: number
  uncertainty_pct: number
}

export interface HistoricalSeriesPoint {
  year: number
  yield: number
}

export interface ForecastYieldResponse {
  state: string
  district: string
  latest_observed_year: number
  latest_observed_yield: number
  historical_series: HistoricalSeriesPoint[]
  forecasts: HorizonForecastItem[]
  model_name: string
  disclaimer: string
}

export interface ForecastYieldRequest {
  state?: string
  district?: string
  horizons?: number[]
}

export interface TrendAnalyzeResponse {
  state: string
  district: string
  linear_slope: number
  theil_sen_slope: number
  mann_kendall_s: number
  p_value: number
  significance: string
  direction: 'STRONG INCREASING' | 'INCREASING' | 'STABLE' | 'DECREASING' | 'STRONG DECREASING'
  observations: number
  first_year: number
  last_year: number
  first_yield: number
  last_yield: number
  total_change_pct: number
  yearly_series: HistoricalSeriesPoint[]
}

export interface StateTrendItem {
  state: string
  state_code: number
  district_count: number
  avg_yield: number
  linear_slope: number
  theil_sen_slope: number
  p_value: number
  significance: string
  direction: string
  total_change_pct: number
}

export interface EarlyWarningAssessResponse {
  state: string
  district: string
  warning_score: number
  severity: 'LOW' | 'MODERATE' | 'HIGH' | 'CRITICAL'
  trend_direction: string
  trend_slope_kg_ha_yr: number
  trend_significance: string
  forecast_1yr_kg_ha: number
  forecast_change_pct: number
  latest_observed_yield: number
  prediction_spread_pct: number
  is_anomaly: boolean
  anomaly_score: number
  trigger_signals: string[]
  components: {
    trend_signal_score: number
    forecast_signal_score: number
    historical_deviation_score: number
    anomaly_signal_score: number
    prediction_spread_score: number
  }
  disclaimer: string
}

export interface EarlyWarningDashboardResponse {
  total_states_monitored: number
  critical_states_count: number
  high_states_count: number
  moderate_states_count: number
  low_states_count: number
  declining_states_count: number
  declining_states: string[]
  average_forecast_spread_pct: number
  top_priority_warnings: EarlyWarningAssessResponse[]
  state_matrix: EarlyWarningAssessResponse[]
}

File: backend/schemas/temporal.py
from typing import List, Optional, Any, Dict
from pydantic import BaseModel, Field

# =========================================================================
# FORECAST SCHEMAS
# =========================================================================

class HorizonForecastItem(BaseModel):
    horizon_years: int
    forecast_year: int
    predicted_yield: float
    lower_bound_p10: float
    upper_bound_p90: float
    prediction_spread: float
    uncertainty_pct: float

class HistoricalSeriesPoint(BaseModel):
    year: int
    yield_val: float = Field(..., alias="yield")

class ForecastYieldRequest(BaseModel):
    state: Optional[str] = Field("Punjab", description="State Name")
    state_code: Optional[int] = Field(None, description="State Code")
    district: Optional[str] = Field(None, description="District Name")
    horizons: List[int] = Field([1, 2, 3], description="Forecast horizons in years")

class ForecastYieldResponse(BaseModel):
    state: str
    district: str
    latest_observed_year: int
    latest_observed_yield: float
    historical_series: List[Dict[str, Any]]
    forecasts: List[HorizonForecastItem]
    model_name: str
    disclaimer: str

# =========================================================================
# TREND ANALYSIS SCHEMAS
# =========================================================================

class TrendAnalyzeRequest(BaseModel):
    state: Optional[str] = Field(None, description="State Name")
    district: Optional[str] = Field(None, description="District Name")

class TrendAnalyzeResponse(BaseModel):
    state: str
    district: str
    linear_slope: float
    theil_sen_slope: float
    mann_kendall_s: float
    p_value: float
    significance: str
    direction: str
    observations: int
    first_year: int
    last_year: int
    first_yield: float
    last_yield: float
    total_change_pct: float
    yearly_series: List[Dict[str, Any]]

class StateTrendItem(BaseModel):
    state: str
    state_code: int
    district_count: int
    avg_yield: float
    linear_slope: float
    theil_sen_slope: float
    p_value: float
    significance: str
    direction: str
    total_change_pct: float

class StatesTrendsResponse(BaseModel):
    data: List[StateTrendItem]

# =========================================================================
# EARLY WARNING SCHEMAS
# =========================================================================

class EarlyWarningAssessRequest(BaseModel):
    state: Optional[str] = Field("Punjab", description="State Name")
    district: Optional[str] = Field(None, description="District Name")

class EarlyWarningAssessResponse(BaseModel):
    state: str
    district: str
    warning_score: float
    severity: str
    trend_direction: str
    trend_slope_kg_ha_yr: float
    trend_significance: str
    forecast_1yr_kg_ha: float
    forecast_change_pct: float
    latest_observed_yield: float
    prediction_spread_pct: float
    is_anomaly: bool
    anomaly_score: float
    trigger_signals: List[str]
    components: Dict[str, float]
    disclaimer: str

class EarlyWarningDashboardResponse(BaseModel):
    total_states_monitored: int
    critical_states_count: int
    high_states_count: int
    moderate_states_count: int
    low_states_count: int
    declining_states_count: int
    declining_states: List[str]
    average_forecast_spread_pct: float
    top_priority_warnings: List[Dict[str, Any]]
    state_matrix: List[Dict[str, Any]]

class StatesEarlyWarningResponse(BaseModel):
    data: List[Dict[str, Any]]
